### PR TITLE
test(core): verify that Ivy i18n works correctly with HTML namespaces

### PR DESCRIPTION
**Pull request by AndrewKushnir**
This commit adds several tests to verify that i18n logic in Ivy handles elements with HTML namespaces correctly.

Related to #36941.

## PR Type
What kind of change does this PR introduce?

- [x] Other... Please describe: test coverage improvement

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/test/acceptance/i18n_spec.ts [hunk 1]
```diff
@@ -530,6 +530,75 @@ onlyInIvy('Ivy i18n logic').describe('runtime i18n', () => {
...
+        ɵsetDocument(document);
+        return document;
+      }
+
+      TestBed.configureTestingModule({
+        providers: [
+          {provide: DOCUMENT, useFactory: _document, deps: []},
+          // TODO(FW-811): switch back to default server renderer (i.e. remove the line below)
+          // once it starts to support Ivy namespace format (URIs) correctly. For now, use
+          // `DomRenderer` that supports Ivy namespace format.
```
Comment (petebacondarwin) [inline packages/core/test/acceptance/i18n_spec.ts:546]: So this is saying that we don't support namespaces on tags when doing SSR?
If so, I agree that this is outside the scope of this PR.

File: packages/core/test/acceptance/i18n_spec.ts [hunk 2]
```diff
@@ -530,6 +530,75 @@ onlyInIvy('Ivy i18n logic').describe('runtime i18n', () => {
...
+            </xhtml:div>
+          </foreignObject>
+        </svg>
+      `);
+
+      const element = fixture.nativeElement;
+      expect(element.textContent.trim()).toBe('Bonjour monde');
+      expect(element.querySelector('svg').namespaceURI).toBe('http://www.w3.org/2000/svg');
+      expect(element.querySelector('div').namespaceURI).toBe('http://www.w3.org/1999/xhtml');
+      expect(element.querySelector('span').namespaceURI).toBe('http://www.w3.org/1999/xhtml');
```
Comment (petebacondarwin) [inline packages/core/test/acceptance/i18n_spec.ts:575]: Interesting... so the `span` inherits its namespace from its containing `div`. I didn't know that. But now it seems obvious.

**Review by petebacondarwin — APPROVED** [review]

File: packages/core/test/acceptance/i18n_spec.ts (hunk 1 above)
Comment (AndrewKushnir) [reply packages/core/test/acceptance/i18n_spec.ts:546]: Hi @petebacondarwin. Yes, this is a pre-existing problem with default server renderer, [see comment here](https://github.com/angular/angular/blob/master/packages/platform-server/src/server_renderer.ts#L76-L78).

**Review by atscott — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
Merge assistance: the only missing approval is `fw-compiler` that I'm a member of. This change is not risky (adding new tests only) and Andrew and Pete LGTM'ed it, so we should be OK to move forward with this change.
